protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/AbstractSenderAuthIdentifyVerificationRcptHook.java

```diff
@@ -68,10 +66,10 @@ private boolean isAnonymous(MaybeSender maybeSender) {
     private boolean senderMatchSessionUser(MaybeSender maybeSender, SMTPSession session) {
         Preconditions.checkArgument(!maybeSender.isNullSender());
 
-        String authUser = session.getUser().toLowerCase(Locale.US);
+        String authUser = session.getUser();
         String username = getUser(maybeSender.get());
 
-        return username.equals(authUser);
+        return username.equalsIgnoreCase(authUser);
```

Comment:
I guess it's not the responsibility of the hook to know is the user repository is case sensitive or not. The user repository should be responsible to check if a username belongs to its database.